File: app/src/main/java/br/com/jonathanzanella/myexpenses/log/LogsView.java
package br.com.jonathanzanella.myexpenses.log;

import android.content.Context;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.AttributeSet;

import org.joda.time.DateTime;

import java.text.SimpleDateFormat;
import java.util.Locale;

import br.com.jonathanzanella.myexpenses.R;
import br.com.jonathanzanella.myexpenses.views.BaseView;
import br.com.jonathanzanella.myexpenses.views.DateTimeView;
import butterknife.Bind;
import butterknife.ButterKnife;

/**
 * Created by jzanella on 7/11/16.
 */
public class LogsView extends BaseView implements DateTimeView.Listener {
	public static final SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yy HH:mm", Locale.getDefault());
	@Bind(R.id.view_logs_list)
	RecyclerView logs;
	@Bind(R.id.view_log_init_time)
	DateTimeView initTime;
	@Bind(R.id.view_log_end_time)
	DateTimeView endTime;
	private LogAdapter adapter;

	public LogsView(Context context) {
		super(context);
	}

	public LogsView(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public LogsView(Context context, AttributeSet attrs, int defStyleAttr) {
		super(context, attrs, defStyleAttr);
	}

	@Override
	protected void init() {
		inflate(getContext(), R.layout.view_logs, this);
		ButterKnife.bind(this);

		adapter = new LogAdapter();

		logs.setAdapter(adapter);
		logs.setLayoutManager(new LinearLayoutManager(getContext()));

		initTime.setDate(DateTime.now().minusHours(1));
		initTime.setListener(this);
		endTime.setDate(DateTime.now());
		endTime.setListener(this);

		adapter.loadData(initTime.getCurrentTime(), endTime.getCurrentTime());
	}

	@Override
	public void onDateTimeChanged(DateTime currentTime) {
		adapter.loadData(initTime.getCurrentTime(), endTime.getCurrentTime());
		adapter.notifyDataSetChanged();
	}
}

File: app/src/main/java/br/com/jonathanzanella/myexpenses/log/LogsView.kt
package br.com.jonathanzanella.myexpenses.log

import android.content.Context
import android.support.v7.widget.LinearLayoutManager
import android.util.AttributeSet
import android.view.View
import android.widget.FrameLayout
import br.com.jonathanzanella.myexpenses.R
import br.com.jonathanzanella.myexpenses.database.RepositoryImpl
import br.com.jonathanzanella.myexpenses.views.BaseView
import br.com.jonathanzanella.myexpenses.views.DateTimeView
import br.com.jonathanzanella.myexpenses.views.FilterableView
import kotlinx.android.synthetic.main.view_logs.view.*
import org.joda.time.DateTime

class LogsView@JvmOverloads constructor(
        context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
) : FrameLayout(context, attrs, defStyleAttr), BaseView, FilterableView, DateTimeView.Listener {

    override var filter = ""
    private val adapter = LogAdapter(LogRepository(RepositoryImpl<Log>(context)))

    init {
        View.inflate(context, R.layout.view_logs, this)

        logs.adapter = adapter
        logs.setHasFixedSize(true)
        logs.layoutManager = LinearLayoutManager(context)
        logs.isNestedScrollingEnabled = false

        initTime.setDate(DateTime.now().minusHours(1))
        initTime.setListener(this)
        endTime.setDate(DateTime.now())
        endTime.setListener(this)

        logLevelError.setOnClickListener { onLogLevel() }
        logLevelWarning.setOnClickListener { onLogLevel() }
        logLevelInfo.setOnClickListener { onLogLevel() }
        logLevelDebug.setOnClickListener { onLogLevel() }

        refreshAdapter()
    }

    override fun onDateTimeChanged(currentTime: DateTime) {
        refreshAdapter()
    }

    fun onLogLevel() {
        refreshAdapter()
    }

    private fun refreshAdapter() {
        adapter.loadData(initTime.currentTime!!, endTime.currentTime!!, getLogLevel(), filter)
        adapter.notifyDataSetChanged()
    }

    private fun getLogLevel(): Log.LogLevel {
        when (logLevel.checkedRadioButtonId) {
            R.id.logLevelError -> return Log.LogLevel.ERROR
            R.id.logLevelWarning -> return Log.LogLevel.WARNING
            R.id.logLevelInfo -> return Log.LogLevel.INFO
            R.id.logLevelDebug -> return Log.LogLevel.DEBUG
        }

        Log.error(LOG_TAG, "new log level?")
        return Log.LogLevel.DEBUG
    }

    override fun filter(s: String) {
        super.filter(s)
        this.filter = s
        refreshAdapter()
    }

    companion object {
        private val LOG_TAG = LogsView::class.java.simpleName
    }
}